pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -170,19 +171,42 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			}
 		}
 		// fire instance handles for workload
-		ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(ip, 0, "", model.AlwaysDiscoverable)
-		workloadInstance := &model.WorkloadInstance{
-			Name:      pod.Name,
-			Namespace: pod.Namespace,
-			Kind:      model.PodKind,
-			Endpoint:  ep,
-			PortMap:   getPortMap(pod),
+		if len(pc.c.handlers.GetWorkloadHandlers()) > 0 {
+			ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(ip, 0, "", model.AlwaysDiscoverable)
+			workloadInstance := &model.WorkloadInstance{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+				Kind:      model.PodKind,
+				Endpoint:  ep,
+				PortMap:   getPortMap(pod),
+			}
+			pc.c.handlers.NotifyWorkloadHandlers(workloadInstance, ev)
 		}
-		pc.c.handlers.NotifyWorkloadHandlers(workloadInstance, ev)
 	}
 	return nil
 }
 
+// checkPodLabelChange fires off eds update if pod label changes
+func (pc *PodCache) checkPodLabelChange(old, cur interface{}) bool {
+	oldPod, ok := old.(*v1.Pod)
+	if !ok {
+		return true
+	}
+	curPod, ok := cur.(*v1.Pod)
+	if !ok {
+		return true
+	}
+	if !labels.Instance(oldPod.Labels).Equals(curPod.Labels) {
```

Comment:
is this better than reflect.DeepEqual

Reply:
I'm not sure but it should be cheaper than reflect, cos reflect has to figure out the type and casting the objects